xcodeproj/xcodeproj.bzl

```diff
@@ -1,307 +1,10 @@
-    root_dirs_file = _write_root_dirs(ctx = ctx)
-    xcodeproj, install_path = _write_xcodeproj(
-        ctx = ctx,
-        project_name = project_name,
-        root_dirs_file = root_dirs_file,
-        spec_file = spec_file,
-    )
-    installer = _write_installer(
-        ctx = ctx,
-        install_path = install_path,
-        xcodeproj = xcodeproj,
-    )
-
-    return [
-        DefaultInfo(
-            executable = installer,
-            files = depset([xcodeproj]),
-            runfiles = ctx.runfiles(files = [xcodeproj]),
-        ),
-        OutputGroupInfo(
-            generated_inputs = generated_inputs,
-        ),
-        XcodeProjOutputInfo(
-            installer = installer,
-            root_dirs = root_dirs_file,
-            spec = spec_file,
-            xcodeproj = xcodeproj,
-        ),
-    ]
-
-def make_xcodeproj_rule(*, transition = None):
-    attrs = {
-        "bazel_path": attr.string(
-            default = "bazel",
-        ),
-        "external_dir_override": attr.string(
-            default = "",
-        ),
-        "generated_dir_override": attr.string(
-            default = "",
-        ),
-        "project_name": attr.string(),
-        "targets": attr.label_list(
-            cfg = transition,
-            mandatory = True,
-            allow_empty = False,
-            aspects = [xcodeproj_aspect],
-        ),
-        "_generator": attr.label(
-            cfg = "exec",
-            # TODO: Use universal generator when done debugging
-            default = Label("//tools/generator:generator"),
-            executable = True,
-        ),
-        "_external_file_marker": attr.label(
-            allow_single_file = True,
-            # This just has to point to a source file in an external repo. It is
-            # only used by a local action, so it doesn't matter what it points
-            # to.
-            default = "@build_bazel_rules_apple//:LICENSE",
-        ),
-        "_install_path": attr.label(
-            default = Label("//xcodeproj/internal:install_path"),
-            providers = [BuildSettingInfo],
-        ),
-        "_installer_template": attr.label(
-            allow_single_file = True,
-            executable = False,
-            default = Label("//xcodeproj/internal:installer.template.sh"),
-        ),
-    }
-
-    if transition:
-        attrs["_allowlist_function_transition"] = attr.label(
-            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
-        )
-
-    return rule(
-        implementation = _xcodeproj_impl,
-        attrs = attrs,
-        executable = True,
-    )
-
-_xcodeproj = make_xcodeproj_rule()
-
-def xcodeproj(*, xcodeproj_rule = _xcodeproj, **kwargs):
-    testonly = kwargs.pop("testonly", True)
+load("//xcodeproj/internal:xcodeproj.bzl", _xcodeproj = "xcodeproj")
 
-    xcodeproj_rule(
-        testonly = testonly,
-        **kwargs
-    )
+xcodeproj = _xcodeproj
+InputFileAttributesInfo = _InputFileAttributesInfo
```

Comment:
Can we add some comments and group them? Here is `rules_swift`'s `swift.bzl`:
```
# Re-export providers.
SwiftInfo = _SwiftInfo
SwiftProtoInfo = _SwiftProtoInfo
SwiftToolchainInfo = _SwiftToolchainInfo
SwiftUsageInfo = _SwiftUsageInfo

# Re-export public API module.
swift_common = _swift_common

# Re-export rules.
swift_binary = _swift_binary
swift_c_module = _swift_c_module
swift_feature_allowlist = _swift_feature_allowlist
swift_grpc_library = _swift_grpc_library
swift_import = _swift_import
swift_library = _swift_library
swift_module_alias = _swift_module_alias
swift_package_configuration = _swift_package_configuration
swift_proto_library = _swift_proto_library
swift_test = _swift_test

# Re-export public aspects.
swift_clang_module_aspect = _swift_clang_module_aspect
swift_usage_aspect = _swift_usage_aspect
```

I know we only have 2, but having the provider right next to the rule looks weird to me 😅.

Reply:
Done.